src/System.Composition.Convention/src/Microsoft/Internal/Requires.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.Reflection;
+
+namespace Microsoft.Internal
+{
+    internal static class Requires
+    {
```

Comment:
Above I say a type called "Assume" that was very similar. Do we need to have both?